### ISSUE TITLE
Test failure: ServerSSLTest.endToEndCloudBackupTest[1] 

### ISSUE THREAD

**Issue by cgtz** [linkedin/ambry#1415]
Fails intermittently on travis. This means that MockNotificationSystem did not have a record of any puts for a specific blob id.
```
com.github.ambry.server.ServerSSLTest > endToEndCloudBackupTest[1] FAILED
    java.lang.NullPointerException
        at com.github.ambry.server.MockNotificationSystem.awaitBlobCreations(MockCluster.java:626)
        at com.github.ambry.server.ServerTestUtil.endToEndCloudBackupTest(ServerTestUtil.java:645)
        at com.github.ambry.server.ServerSSLTest.endToEndCloudBackupTest(ServerSSLTest.java:143)
```
Build scan: https://scans.gradle.com/s/7exzgoidx3ymm/tests/6awkxu3uswmr6-fd54lkcvnxxmm

**Comment by cgtz** [comment]
```
com.github.ambry.server.ServerSSLTest > endToEndCloudBackupTest[1] FAILED
    java.lang.NullPointerException
        at com.github.ambry.server.MockNotificationSystem.awaitBlobCreations(MockCluster.java:636)
        at com.github.ambry.server.ServerTestUtil.endToEndCloudBackupTest(ServerTestUtil.java:659)
        at com.github.ambry.server.ServerSSLTest.endToEndCloudBackupTest(ServerSSLTest.java:143)
```
https://gradle.com/s/vnvyw7zk4jcyk

**Comment by lightningrob** [comment]
I am able to reproduce this locally by setting a breakpoint in MockNotificationSystem.onBlobReplicaCreated().  The main thread that calls:
    assertTrue("Did not put all blobs in 2 minutes", latch.await(2, TimeUnit.MINUTES));
still completes successfully and proceeds to check   notificationSystem.awaitBlobCreations() and get the NPE.

I think the reason it completes is that the latch value (1) that is passed to the DirectSender class does not match the number of blobs (10):

    int blobBackupCount = 10;
    CountDownLatch latch = new CountDownLatch(1);
    DirectSender runnable =
        new DirectSender(cluster, channel, blobBackupCount, data, userMetadata, properties, null, latch);

Note that the test method endToEndReplicationWithMultiNodeMultiPartitionTest() has the same issue.

**Comment by lightningrob** [comment]
The above comment might be incorrect.  DirectRunner calls latch.countDown() only once after all blobs are put.

**Comment by lightningrob** [comment]
This should be fixed by #1654
